infrastructure/network/connmanager/outgoing_connections.go

```diff
@@ -24,46 +28,26 @@ func (c *ConnectionManager) checkOutgoingConnections(connSet connectionSet) {
 
 	connectionsNeededCount := c.targetOutgoing - len(c.activeOutgoing)
 	connectionAttempts := connectionsNeededCount * 2
-	for i := 0; i < connectionAttempts; i++ {
+	netAddresses := c.addressManager.RandomAddresses(connectionAttempts, connectedAddresses)
+
+	for _, netAddress := range netAddresses {
 		// Return in case we've already reached or surpassed our target
 		if len(c.activeOutgoing) >= c.targetOutgoing {
```

Comment:
Since we no longer do the `* 2` (see separate comment above), we no longer need this clause.